Add Document controller documentation (#32)
kuzzleio/sdk-csharp: doc/1/controllers/document/create/index.md

---
code: true
type: page
title: CreateAsync
description: Creates a new document.
---

# CreateAsync

Creates a new document in the persistent data storage.

> Throws an error if the document already exists.

Reviewer: Misleading:

```suggestion
Throws an error if an ID is provided and if a document with that ID already exists.
```

## Arguments

```csharp
public async Task<JObject> CreateAsync( 
  string index, 
  string collection, 
  JObject content, 
  string id = null, 
  bool waitForRefresh = false);

```

<br/>

| Option     | Type<br/>(default)                       | Description                                                                        |
| ------------ | ------------------------------------ | ----------------------------------------------------------- |
| `index`      | <pre>string</pre>        | Index name                                                  |
| `collection` | <pre>string</pre>        | Collection name                                             |

> | `content`   | <pre>string</pre>        | JObject representing the body of the document           |

Reviewer 2: ```suggestion
| `content`   | <pre>JObject</pre>        | JObject representing the body of the document           |
```

> | `id`         | <pre>string</pre><br/>(`null`)        | Document ID. Will use an auto-generated id if not specified |

Reviewer: ```suggestion
| `id`         | <pre>string</pre><br/>(`null`)        | Document ID. Will use an auto-generated ID if not specified |
```
